.github/workflows/ci.yml

```diff
@@ -78,43 +78,114 @@ jobs:
+          fi
+          
+          # Make sure it's in PATH
+          if [[ "${{ matrix.platform.os }}" == windows* ]]; then
+            if [ -f "p2pdCache/p2pd.exe" ]; then
+              echo "${{ github.workspace }}/p2pdCache" >> $GITHUB_PATH
+            elif [ -f "p2pd.exe" ]; then
+              # Copy to p2pdCache if it was built in a different location
+              mkdir -p p2pdCache
+              cp p2pd.exe p2pdCache/
+              echo "${{ github.workspace }}/p2pdCache" >> $GITHUB_PATH
+            fi
+          else
+            if [ -f "p2pdCache/p2pd" ]; then
+              chmod +x p2pdCache/p2pd
+              echo "${{ github.workspace }}/p2pdCache" >> $GITHUB_PATH
+            elif [ -f "p2pd" ]; then
+              # Copy to p2pdCache if it was built in a different location
+              mkdir -p p2pdCache
+              cp p2pd p2pdCache/
+              chmod +x p2pdCache/p2pd
+              echo "${{ github.workspace }}/p2pdCache" >> $GITHUB_PATH
+            fi
+          fi
+          
+          # Ensure executable permissions
+          if [[ "${{ matrix.platform.os }}" != windows* ]]; then
+            chmod +x p2pdCache/p2pd || true
+          fi
+          
+          # Verify it's working
+          export PATH="${{ github.workspace }}/p2pdCache:$PATH"
+          if [[ "${{ matrix.platform.os }}" == windows* ]]; then
+            p2pdCache/p2pd.exe --version || echo "p2pd version check failed"
+          else
+            p2pdCache/p2pd --version || echo "p2pd version check failed"
+          fi
 
+      # Rest of the CI steps
       - name: Restore deps from cache
         id: deps-cache
         uses: actions/cache@v3
         with:
           path: nimbledeps
-          # Using nim.ref as a simple way to differentiate between nimble using the "pkgs" or "pkgs2" directories.
-          # The change happened on Nimble v0.14.0. Also forcing the deps to be reinstalled on each os and cpu.
-          key: nimbledeps-${{ matrix.nim.ref }}-${{ matrix.builder }}-${{ matrix.platform.cpu }}-${{ hashFiles('.pinned') }} # hashFiles returns a different value on windows
+          key: nimbledeps-${{ matrix.nim.ref }}-${{ matrix.builder }}-${{ matrix.platform.cpu }}-${{ hashFiles('.pinned') }}
+
+      - name: Setup python
+        run: |
+          mkdir -p .venv
+          python -m venv .venv
 
       - name: Install deps
         if: ${{ steps.deps-cache.outputs.cache-hit != 'true' }}
         run: |
           nimble install_pinned
 
       - name: Use gcc 14
-        if : ${{ matrix.platform.os == 'linux-gcc-14'}}
+        if: ${{ matrix.platform.os == 'linux-gcc-14' }}
         run: |
           # Add GCC-14 to alternatives
           sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-14 14
-
           # Set GCC-14 as the default
           sudo update-alternatives --set gcc /usr/bin/gcc-14
 
       - name: Run tests
         run: |
+          source .venv/bin/activate
```

Comment:
This is no longer needed. I ended up removing python from mbedtls (and ended up remobing mbedtls too :) )